lib/cast_vote_msg.js

```diff
@@ -1,22 +1,21 @@
 const Assert = require('assert-plus')
 const Vote = require('./vote_entity')
 const Shapes = require('./shapes')
-const { fetchProp } = require('./utils')
-const { ValidationError, NotFoundError } = require('./errors')
+const { NotFoundError } = require('./errors')
 const VoteStats = require('./vote_stats')
 const PollRating = require('./poll_rating')
 const Reply = require('./reply')
 
-module.exports = function (opts = {}) {
+module.exports = function (opts) {
   async function handler(msg, reply) {
     try {
-      const vote_type = fetchProp(msg, 'vote')
+      const vote_type = msg.vote
       const safe_params = await validateMessage(msg)
-      const poll_id = fetchProp(safe_params, ['fields', 'poll_id'])
-      const voter_id = fetchProp(safe_params, ['fields', 'voter_id'])
-      const voter_type = fetchProp(safe_params, ['fields', 'voter_type'])
-      const vote_kind = fetchProp(safe_params, ['fields', 'kind'])
-      const vote_code = fetchProp(safe_params, ['fields', 'code'])
+      const poll_id = safe_params.fields.poll_id
```

Comment:
to avoid name repetition, how about using your idea:

const { fields } = safe_params
const { voter_id } = fields
etc